### ISSUE TITLE
StackOverflow on running unit tests

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#6089]
VS Mac 7.7.0.1424 (release-7.7)

To repro:

1. Open Main.sln (monodevelop repo).
2. Open Unit Tests window.
3. Expand Main - Addins - AspNet.
4. Right click AspNet and select Run Test.

(Right clicking the MonoDevelop.AspNet.Tests tree node below AspNet is OK).

```
Stack overflow: IP: 0x15b6182b8, fault addr: 0x7ffeedce9ff8
Stacktrace:
  at MonoDevelop.Projects.SolutionFolder/<GetExecutionDependencies>d__31.<>m__Finally2 () [0x00016] in <73b0baa33214478cad62946dad197e80>:0
  <...>
  at MonoDevelop.Projects.SolutionFolder/<GetExecutionDependencies>d__31.System.IDisposable.Dispose () [0x00022] in <73b0baa33214478cad62946dad197e80>:0
  at MonoDevelop.Projects.SolutionFolder/<GetExecutionDependencies>d__31.<>m__Finally2 () [0x00016] in <73b0baa33214478cad62946dad197e80>:0
....
/Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SolutionFolder.cs:418
  at MonoDevelop.Projects.SolutionFolder/<GetExecutionDependencies>d__31.MoveNext () [0x000d4] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SolutionFolder.cs:418
....
  at System.Linq.Enumerable/SelectManySingleSelectorIterator`2<TSource_REF, TResult_REF>.MoveNext () [0x0006f] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/SelectMany.cs:209
  at MonoDevelop.Ide.ProjectOperations/SolutionItemBuildBatch.Create (System.Collections.Generic.IEnumerable`1<MonoDevelop.Projects.IBuildTarget>) [0x00080] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.SolutionItemBuildBatch.cs:54
  at MonoDevelop.Ide.ProjectOperations/<CheckAndBuildForExecute>d__109.MoveNext () [0x001bb] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.cs:1445
  at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<bool>.Start<MonoDevelop.Ide.ProjectOperations/<CheckAndBuildForExecute>d__109> (MonoDevelop.Ide.ProjectOperations/<CheckAndBuildForExecute>d__109&) [0x0002c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:471
  at MonoDevelop.Ide.ProjectOperations.CheckAndBuildForExecute (System.Collections.Generic.ICollection`1<MonoDevelop.Projects.IBuildTarget>,MonoDevelop.Projects.ConfigurationSelector,bool,bool,System.Func`3<MonoDevelop.Projects.IBuildTarget, MonoDevelop.Core.ProgressMonitor, System.Threading.Tasks.Task>,System.Nullable`1<System.Threading.CancellationToken>) [0x0005a] in <1d0e2ef745fb4d9e8b590256c66a4f8d>:0
  at MonoDevelop.UnitTesting.UnitTestService/<RunTests>d__10.MoveNext () [0x0007b] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestService.cs:133
  at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<MonoDevelop.UnitTesting.UnitTestService/<RunTests>d__10> (MonoDevelop.UnitTesting.UnitTestService/<RunTests>d__10&) [0x0002c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:316
  at MonoDevelop.UnitTesting.UnitTestService.RunTests (System.Collections.Generic.IEnumerable`1<MonoDevelop.UnitTesting.UnitTest>,MonoDevelop.Projects.ExecutionContext,bool,bool,System.Threading.CancellationTokenSource) [0x00048] in <d5706bc2e7f0446691715e02278dcdcf>:0
  at MonoDevelop.UnitTesting.UnitTestService.RunTests (System.Collections.Generic.IEnumerable`1<MonoDevelop.UnitTesting.UnitTest>,MonoDevelop.Projects.ExecutionContext,bool) [0x00006] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestService.cs:181
  at MonoDevelop.UnitTesting.UnitTestService.RunTests (System.Collections.Generic.IEnumerable`1<MonoDevelop.UnitTesting.UnitTest>,MonoDevelop.Projects.ExecutionContext) [0x00000] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestService.cs:173
  at MonoDevelop.UnitTesting.TestPad.RunTests (System.Collections.Generic.IEnumerable`1<MonoDevelop.UnitTesting.UnitTest>,MonoDevelop.Core.Execution.IExecutionHandler,bool) [0x00070] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Gui/TestPad.cs:541
  at MonoDevelop.UnitTesting.TestPad.RunTests (MonoDevelop.Ide.Gui.Components.ITreeNavigator[],MonoDevelop.Core.Execution.IExecutionHandler,bool) [0x0004e] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Gui/TestPad.cs:516
  at MonoDevelop.UnitTesting.TestPad.RunSelectedTest (MonoDevelop.Core.Execution.IExecutionHandler) [0x00000] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Gui/TestPad.cs:553
  at MonoDevelop.UnitTesting.TestPad.OnRunTest () [0x00000] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Gui/TestPad.cs:412
  at (wrapper runtime-invoke) object.runtime_invoke_void__this__ (object,intptr,intptr,intptr) [0x0004f] in <016f22b59bd64951b06e38b9f5a0506c>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) [0x00016] in <016f22b59bd64951b06e38b9f5a0506c>:0
  at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) [0x0003b] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/MonoMethod.cs:305
  at System.Reflection.MethodBase.Invoke (object,object[]) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/reflection/methodbase.cs:229
  at MonoDevelop.Components.Commands.CommandHandlerInfo.Run (object,MonoDevelop.Components.Commands.Command) [0x00022] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:2572
  at MonoDevelop.Components.Commands.CommandManager/<>c__DisplayClass115_3.<DispatchCommand>b__1 () [0x00075] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:1523
  at MonoDevelop.Components.Commands.CommandManager.DispatchCommand (object,object,object,MonoDevelop.Components.Commands.CommandSource,System.Nullable`1<uint>,MonoDevelop.Components.Commands.CommandInfo) [0x00387] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:1542
  at MonoDevelop.Components.Commands.CommandManager.DispatchCommand (object,object,object,MonoDevelop.Components.Commands.CommandSource,MonoDevelop.Components.Commands.CommandInfo) [0x00000] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:1419
  at MonoDevelop.Components.Mac.MDMenuItem/<>c__DisplayClass14_0.<Run>b__0 () [0x0005c] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Mac/MDMenuItem.cs:90
  at GLib.Timeout/TimeoutProxy.HandlerInternal (intptr) [0x00017] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/gtk-sharp-None/glib/Timeout.cs:38
  at (wrapper native-to-managed) GLib.Timeout/TimeoutProxy.HandlerInternal (intptr) [0x00021] in <bbc097c736e649f0bbe49b81946fd886>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) Gtk.Application.gtk_main () [0x00008] in <13dfc6c534074eb08a8f382a81e605d5>:0
  at Gtk.Application.Run () [0x00001] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:145
  at MonoDevelop.Ide.IdeApp.Run () [0x00006] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/Ide.cs:464
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions) [0x009d4] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:377
  at MonoDevelop.Ide.IdeStartup.Main (string[],MonoDevelop.Ide.Extensions.IdeCustomizer) [0x000b2] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:804
  at Xamarin.Startup.MainClass.Main (string[]) [0x00000] in /Users/vsts/agent/2.140.2/work/1/s/md-addins/Xamarin.Startup/Main.cs:11
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) [0x00057] in <ac8fbde34861483b842fb6311d5c9e7d>:0

```

> VS bug [#691711](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/691711)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.657 (master)

Pull Request #6123 merged by: Lluis Sanchez
Author: mono
Commit: 51b3aff255797222595e41ed643a0e8dfb62e02b (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1487 (release-7.7)

Pull Request #6124 merged by: Lluis Sanchez
Author: mono
Commit: 8bacc2dbe315fe6349ac0871a953afb0e0f127e7 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.197 (release-8.0)

Pull Request #6123 merged by: Lluis Sanchez
Author: mono
Commit: 51b3aff255797222595e41ed643a0e8dfb62e02b (mono/monodevelop)
